torch/_dynamo/utils.py

```diff
@@ -751,6 +751,23 @@ def dict_const_keys(value):
 def global_key_name(key):
     return f"__dict_key_{id(key)}"
 
+def dup_guard_decayed_equality_debug(t1, t2):
+    if t1 is t2:
+        print(f"{t1} is {t2}")
+        return True
+    
+    hack_set = set()
+    hack_set.add(t1)
+    
+    if t2 in hack_set:
+        print(f"{t1} is not {t2}, but hashes the same")
+        return True
+    if torch.equal(t1, t2):
```

Comment:
wait, I don't think the `is` guards can be relaxed to `tensors contain same values`

Reply:
This isn't meant to land at all, this is to help some folks debug